qutebrowser/completion/models/histcategory.py

```diff
@@ -90,13 +90,8 @@ def set_pattern(self, pattern):
             self._query.run(pat=pattern)
         self.setQuery(self._query)
 
-    def delete_cur_item(self, index):
-        """Delete the row at the given index."""
-        if not self.delete_func:
-            raise cmdexc.CommandError("Cannot delete this item.")
-        data = [self.data(index.sibling(index.row(), i))
-                for i in range(self.columnCount())]
-        self.delete_func(data)
+    def removeRow(self, _row, _col):
```

Comment:
There's [QAbstractItemModel::removeRow](http://doc.qt.io/qt-5/qabstractitemmodel.html#removeRow) already, which mentions that it calls `removeRows`. Also, `removeRow` is not `virtual` in C++, so if this gets called by Qt internally for some reason, it wouldn't use the overloaded version - so I think it'd be better to implement `removeRows` and then use `removeRow` without overloading that.